app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -86,8 +86,19 @@ EOF
 elif [ -f /etc/debian_version ]; then
   <%= save_to_file('/etc/apt/sources.list.d/foreman_registration.list', @repo) %>
 <% if @repo_gpg_key_url.present? -%>
-  apt-get -y install ca-certificates gpg
-  curl --silent --show-error --output /etc/apt/trusted.gpg.d/client.asc <%= shell_escape @repo_gpg_key_url %>
+<% # "apt 1.2.35" on Ubuntu 16.04 does not support storing GPG public keys in "/etc/apt/trusted.gpg.d/" in ASCII format -%>
+<%
+  use_gpg_dearmor = @host.operatingsystem.name == 'Ubuntu' && @host.operatingsystem.major.split('.')[0].to_i < 18
+-%>
+
+  <% if use_gpg_dearmor -%>
+    apt-get -y install curl ca-certificates pgpgpg
```

Comment:
gpg, not pgpgpg

Reply:
On Ubuntu 16.04, the package is called "pgpgpg":

```
$ podman run --rm -it ubuntu:xenial

# apt-get update
...
# apt-get install gpg
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package gpg is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'gpg' has no installation candidate

# apt-get install pgpgpg
...
The following NEW packages will be installed:
  pgpgpg
...

# gpg --version
gpg (GnuPG) 1.4.20
...
```

Reply:
Looking at https://packages.ubuntu.com/focal/pgpgpg it's a wrapper that depends on gnupg so I'd expect that gnupg is sufficient

Reply:
I have changed the name of the package.